### PR TITLE
UI: Added `DraggableHeader`, `ResizeablePanel` and a basic menu

### PR DESCRIPTION
**Pull request by Alf21**
Waiting for:
- https://github.com/Facepunch/sbox-issues/issues/535
- https://github.com/Facepunch/sbox-issues/issues/536

https://user-images.githubusercontent.com/11843790/127728154-a12c7adb-55d8-4f55-9d0b-5647d0ad4ebe.mp4

image

To test:
- Run `ttt_menu` via console

### CODE REVIEW

File: code/player/TTTPlayer.Commands.Client.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+namespace TTTReborn.Player
+{
+    public partial class TTTPlayer
+    {
+        [ClientCmd(Name = "ttt_playerids", Help = "Returns a list of all players (clients) and their associated IDs")]
+        public static void PlayerID()
+        {
+            List<Client> playerList = Client.All.ToList();
+
+            for (int i = 0; i < playerList.ToList().Count; i++)
```
Comment (mzegar) [inline code/player/TTTPlayer.Commands.Client.cs:15]: We are doing `ToList()` twice here?

File: code/ui/richpanel/RichPanel.Resizeable.cs [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+                    }
+                    else
+                    {
+                        CurrentDragAnchor = value;
+                    }
+                }
+                else
+                {
+                    CurrentDragAnchor = CurrentVerticalDragAnchor;
+                }
```
Comment (mzegar) [inline code/ui/richpanel/RichPanel.Resizeable.cs:95]:
```suggestion
                _currentHorizontalDragAnchor = value ?? CurrentVerticalDragAnchor;

                    if (CurrentDragAnchor == DragAnchor.TOP)
                    {
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.TOPLEFT : DragAnchor.TOPRIGHT;
                    }
                    else if (CurrentDragAnchor == DragAnchor.BOTTOM)
                    {
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.BOTTOMLEFT : DragAnchor.BOTTOMRIGHT;
                    }
                    else
                    {
                        CurrentDragAnchor = value;
                    }
```

Can this work, haven't taken an in-depth look.

File: code/ui/richpanel/RichPanel.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+namespace TTTReborn.UI
+{
+    public partial class RichPanel : TTTPanel
+    {
+        public RichPanel() : base()
+        {
+            StyleSheet.Load("/ui/richpanel/RichPanel.scss");
+
+            //AcceptsFocus = true;
```
Comment (mzegar) [inline code/ui/richpanel/RichPanel.cs:9]: Need this comment?

File: code/ui/tttpanel/TTTPanel.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            }
+        }
+        private bool _isShowing = false;
+
+        public TTTPanel() : base()
+        {
+            IsShowing = false;
+        }
+    }
+}
```
Comment (mzegar) [inline code/ui/tttpanel/TTTPanel.cs:24]: Nice :D, do you want to add an issue whenever this is merged to add this to all panels?

File: code/ui/richpanel/RichPanel.cs (hunk 3 above)
Comment (Alf21) [reply code/ui/richpanel/RichPanel.cs:9]: Yea as this is a next step to catch shortcuts :)

File: code/ui/tttpanel/TTTPanel.cs (hunk 4 above)
Comment (Alf21) [reply code/ui/tttpanel/TTTPanel.cs:24]: You mean part of a refactoring like the still pending https://github.com/TTTReborn/tttreborn/pull/139 :P

File: code/ui/richpanel/RichPanel.Resizeable.cs (hunk 2 above)
Comment (mzegar) [reply code/ui/richpanel/RichPanel.Resizeable.cs:95]: nvm sec, let me throw up a better snippet.

Comment (Alf21) [reply code/ui/richpanel/RichPanel.Resizeable.cs:95]: Please test these changes :) If I'm looking into it, I think last line in the last else should be `CurrentDragAnchor 
 = _currentHorizontalDragAnchor;`

File: code/player/TTTPlayer.Commands.Client.cs (hunk 1 above)
Comment (Alf21) [reply code/player/TTTPlayer.Commands.Client.cs:15]: lol, yea just put it into another file :D

File: code/ui/tttpanel/TTTPanel.cs (hunk 4 above)
Comment (Alf21) [reply code/ui/tttpanel/TTTPanel.cs:24]: https://github.com/TTTReborn/tttreborn/issues/168

File: code/ui/richpanel/RichPanel.Resizeable.cs (hunk 2 above)
Comment (mzegar) [reply code/ui/richpanel/RichPanel.Resizeable.cs:95]:
```csharp
        public DragAnchor? CurrentHorizontalDragAnchor
        {
            get
            {
                return _currentHorizontalDragAnchor;
            }
            private set
            {
                _currentHorizontalDragAnchor = value ?? CurrentVerticalDragAnchor;

                switch (CurrentDragAnchor)
                {
                    case DragAnchor.TOP:
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.TOPLEFT : DragAnchor.TOPRIGHT;
                        break;
                    case DragAnchor.BOTTOM:
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.BOTTOMLEFT : DragAnchor.BOTTOMRIGHT;
                        break;
                    default:
                        CurrentDragAnchor = _currentHorizontalDragAnchor;
                        break;
                }
            }
        }
        private DragAnchor? _currentHorizontalDragAnchor = null;
```

Comment (Alf21) [reply code/ui/richpanel/RichPanel.Resizeable.cs:95]: But isn't the switch running even if `value` is null? So what if `CurrentDragAnchor` is `DragAnchor.TOP`? Then we would get `DragAnchor.TOPRIGHT` i think.

**Review by mzegar — APPROVED** [review]
LGTM

**Pull request merged**